### PR TITLE
Fix(content-manager): Clicking a filter deletes the filter instead of allowing to edit it.

### PR DESCRIPTION
**Pull request by moeez506**
### What does it do?

Filter edit functionality added. Clicking a filter other than the "x" will allow you to edit that filter.

### Why is it needed?

Clicking a filter anywhere deletes it rather than allowing you to edit it

### How to test it?

Navigate to content manager and add any filter then click on it, will allow you to edit.

### Related issue(s)/PR(s)
#20967 

### Need Assitance. 
 I have solved 90% issue but the design part is ambigous, right now the edit modal is stick to the top left corner which should be exactly below the filter being edited. Needed some guidance regarding that. 
 Attached recording for refrence 
 https://www.loom.com/share/07f37672aaeb4860b3eefe14c53cdf03?sid=40084628-af47-4d91-a237-0c201da496f5
 
 PS: New to strapi codebase, it's my first contribution here.

### CODE REVIEW

File: packages/core/helper-plugin/src/components/FilterListURLQuery.tsx [hunk 1]
```diff
@@ -162,8 +186,17 @@ const AttributeTag = ({ attribute, filter, onClick, operator, value }: Attribute
   return (
     <Box padding={1}>
       <Tag onClick={handleClick} icon={<Cross />}>
-        {content}
+        <span onClick={handleContentClick}>{content}</span>
```
Comment (simotae14) [inline packages/core/helper-plugin/src/components/FilterListURLQuery.tsx:189]: please, span is a text, its behaviour is not to have an onClick, if you want to handle the onClick in different way you can add a condition in the handleClick handler, there is not need of two handlers for the same thing, you can differentiate the behaviour base on the event target element that triggered the handleClick for example

File: packages/core/helper-plugin/src/components/FilterListURLQuery.tsx [hunk 2]
```diff
@@ -162,8 +186,17 @@ const AttributeTag = ({ attribute, filter, onClick, operator, value }: Attribute
...
       </Tag>
+      {isVisible && (
+        <FilterPopoverURLQuery
+          displayedFilters={filtersSchema}
+          isVisible={true}
+          onToggle={handleToggle}
+          source={buttonRef}
+          initialFilter={filter}
+        />
+      )}
```
Comment (simotae14) [inline packages/core/helper-plugin/src/components/FilterListURLQuery.tsx:199]: to change the position of this dropdown you need to assign the buttonRef to a button but in your case you are passing it just to the FilterPopoverURLQuery component
Please take a look how the FilterPopoverURLQuery is used in other places for example in this file packages/core/admin/admin/src/pages/Settings/components/Filters.tsx

**Review by simotae14 — CHANGES_REQUESTED** [review]
Hello @moeez506, thanks a lot for your contribution,
please take a look at my comments and before submitting your code check locally if it passed all the necessary steps (run prettier command, unit and e2e tests, build the package you changed), thanks in advance

**Comment by alexandrebodin** [comment]
HI @moeez506 we will be closing this PR as staled but if you eventually have time to work on it if we haven't done it in the future feel free to ping us to reopen 🙏  

Thank you for contributing!

**Pull request closed without merging**